fix(css): Clarify text and example for complex selectors with general sibling combinator (~) (#29093)
mdn/content: files/en-us/web/css/general_sibling_combinator/index.md

```css-nolint
/* The white space around the ~ combinator is optional but recommended. */

> compound_selector1 ~ compound_selector2 { style properties }

Reviewer: it doesn't have to be a compound selector. i think the original is easier to understand here. And the first example uses simple selectors which makes the original more accurate.

Author: Got it. Reverted to original

```

## Examples

### Using the combinator with simple selectors

> This example shows the use of the `~` combinator when both the selectors are simple (`p` and `span`).

Reviewer: ```suggestion
This example shows the use of the `~` combinator when both the selectors are simple selectors (`p` and `span`).
```

  </span>
</article>
<span>
  This span is not red because it doesn't share a parent with a paragraph.
</span>
```

```css
p ~ span {
  color: red;
}
```

{{EmbedLiveSample("Using the combinator with simple selectors", "auto", 300)}}

### Using the combinator with compound selectors

> Elements represented by the two compound selectors share the same parent in the document tree. The element represented by the first compound selector precedes (not necessarily immediately) the element represented by the second one.

Reviewer: ```suggestion
In this example, elements represented by the two compound selectors share the same parent in the document tree. The element represented by the first compound selector precedes (not necessarily immediately) the element represented by the target selector.
```

> In this example, elements represented by the two [complex selectors](/en-US/docs/Web/CSS/CSS_selectors/Selector_structure#complex_selector) share the same parent in the document tree. The element represented by the first complex selector precedes (not necessarily immediately) the element represented by the target selector.

Reviewer: this is a bit confusing and not pedantically accurate. The span (simple selector) needs to be a sibling of the p that happens to be a descendant of .foo. The span is a sibling of p. both simple selectors. Similarly, the second on, is "all spans that are descendants of .foo, if that foo is a sibling of p that is a descendant of foo. I guess what I am saying is the whole thing is a complex selector, it's not a complex selector before and after the ~, which i think is what this is saying. Maybe:
```suggestion
This example demonstrates two [complex selectors](/en-US/docs/Web/CSS/CSS_selectors/Selector_structure#complex_selector). The first complex selector matches all spans occurring after a paragraph in the document tree if the span and paragraph share the same parent if that parent or an ancestor of that parent has the class `.foo`.  In the second complex selector, the span being matched is a descendant of element `.foo` if that element is a sibling of the previously mentioned paragraph.
```

> The example below shows that the second compound selector must match the parent of the first compound selector.

Reviewer: ```suggestion
The example below shows that the target compound selector must share the same parent of the first compound selector.
```

```html
<h1>Dream big</h1>
<span>And yet again this is a red span!</span>
<div class="foo">
  <p>Here is another paragraph.</p>
  <span>A blue span</span>
  <div class="foo">
    <span>A green span</span>
  </div>
</div>
```

```css
.foo p ~ span {
  color: blue;
}

.foo p ~ .foo span {
  color: green;
}
```

{{EmbedLiveSample("Using the combinator with complex selectors", "auto", 200)}}

> In the above HTML, the two siblings of `.foo p` are `span` and `.foo span`.

Reviewer: ```suggestion
In the above HTML, the two siblings of `.foo p` are `span` and `.foo`. The green span is a descendant of the `foo` that is a sibling of `p`.
```

Author: Thank you!!

Just to be sure, should it be "sibling of `.foo p`" at the end, like so:

> In the above HTML, the two siblings of `.foo p` are `span` and `.foo`. The green `span` is a descendant of `.foo`, which is a sibling of `.foo p`.

Reviewer: it can be. there are so many ways to verbalize it. It could also be:I

```
In the above HTML, the two siblings of .foo p are span and .foo. The green span is a descendant of .foo. This `.foo` is a sibling of `p`, with both being children of the parent `<div> with the same `.foo` class.
```

Author: Thanks! I went with you original suggestion.

> - When the second compound selector is `span`, it selects the `span` element that is a sibling of `p` and both are descendants of `.foo`.

Reviewer: ```suggestion
- When the target selector is `span`, it selects the `span` element that is a sibling of `p`. As  `p` is a descendant of `.foo`, so are its `span` siblings.
```

> - When the second compound selector is `.foo span`, it selects the `span` element that's a descendant of `.foo` if that `.foo` is a sibling of `p`; essentially, both are nested in an ancestor of `.foo`.

Reviewer: ```suggestion
- In `.foo p ~ .foo span`, the target selector is `span` that is a descendant of `.foo`. In this case, it selects the `span` element that's a descendant of `.foo` if that `.foo` is a sibling of `p`; essentially, both are nested in an ancestor of `.foo`.
```

`.foo span` is a complex selector, not a compound selector. The above that i wrote is a bit redundant, but i think it's ok.

Author: Thanks for the hints!

## See also

- [Adjacent sibling combinator](/en-US/docs/Web/CSS/Adjacent_sibling_combinator)

> -

Reviewer: ```suggestion
```
